tests/baselines/reference/quickInfoForJSDocWithHttpLinks.baseline

```diff
@@ -286,7 +286,18 @@
               "kind": "link"
             },
             {
-              "text": "https://hva",
+              "text": "https",
```

Comment:
This doesn't look like an improvement?

Reply:
It actually is; above, the test has a `@typedef {...} https://wot`, which defines a type named `https`. The text rendered here is still 'https://hva'.

Reply:
The first line of the test is:

```ts
//// /** @typedef {number} /*1*/https://wat */
```

Reply:
I'm extremely skeptical that anyone would write that and mean anything other than that they wanted a link to a URL.  What's the editor experience now?  "https" is a clickable link to the typeref above and "://hva" is plain text?  And quick info shows it as though it were a URL?  There's no space or formatting change between the parts?

Reply:
I can imagine someone thinking they were clever and writing `@typedef {string} https` so that they could flag strings containing URLs, but I can't imagine them then wanting `{@link https://www.bing.com}` to refer to that typedef.

Reply:
If the editor doesn't override the display parts formatting, then `https` is a clickable link and `://hva` is non-clickable test not separated by spaces. 

1. In `master`, you can repro this by defining `class https` instead of `type https`, so it's not related to this PR.
2. My intention for `@link` support is primarily to link to in-compilation entities, not URLs. People use markdown links for that already (and Code supports it).

If the editor experience breaks when `https` is in scope, the author can remove that declaration or switch to using the markdown format for links. I think the current behaviour is fine and this test of a corner case shows it off.